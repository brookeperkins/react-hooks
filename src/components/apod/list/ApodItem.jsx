import React from 'react';
import PropTypes from 'prop-types';

const ApodItem = ({ title, url }) => (
  <figure>
    <h1>{title}</h1>
    <img src={url} alt={title} />
  </figure>
);

ApodItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default ApodItem;
