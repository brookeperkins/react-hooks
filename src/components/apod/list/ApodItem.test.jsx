import React from 'react';
import { render } from '@testing-library/react';
import ApodItem from './ApodItem';
/* eslint-disable max-len */

describe('ApodItem component', () => {
  it('renders ApodItem component', () => {
    const { asFragment } = render(<ApodItem 
      title="NGC 6369: The Little Ghost Nebula"
      url="https://apod.nasa.gov/apod/image/1710/NGC6369LittleGhostHST1024c.jpg"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
