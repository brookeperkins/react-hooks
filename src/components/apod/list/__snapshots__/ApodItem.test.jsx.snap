// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApodItem component renders ApodItem component 1`] = `
<DocumentFragment>
  <figure>
    <h1>
      NGC 6369: The Little Ghost Nebula
    </h1>
    <img
      alt="NGC 6369: The Little Ghost Nebula"
      src="https://apod.nasa.gov/apod/image/1710/NGC6369LittleGhostHST1024c.jpg"
    />
  </figure>
</DocumentFragment>
`;
